torch_geometric/data/dataset.py

```diff
@@ -80,10 +80,12 @@ def __init__(self, root: Optional[str] = None,
         self.pre_filter = pre_filter
         self._indices: Optional[Sequence] = None
 
-        if self.download.__qualname__.split('.')[0] != 'Dataset':
+        download_method = getattr(self, "download", None)
+        if callable(download_method):
             self._download()
 
-        if self.process.__qualname__.split('.')[0] != 'Dataset':
+        process_method = getattr(self, "process", None)
```

Comment:
This breaks the fix. I think it basically goes back to the original implementation because `getattr(self, "process", None)` will never be `None`?
```python
callable(getattr(torch_geometric.data.Dataset(), "download", None))
> True
```